extensions/Worker.Extensions.CosmosDB/src/Extensions/FunctionsWorkerApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Provides extension methods to work with a <see cref="IFunctionsWorkerApplicationBuilder"/>.
+    /// </summary>
+    public static class FunctionsWorkerApplicationBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the CosmosDB extension for FSharp functions.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFunctionsWorkerApplicationBuilder"/> to configure.</param>
+        /// <returns>The same instance of the <see cref="IFunctionsWorkerApplicationBuilder"/> for chaining.</returns>
+        public static IFunctionsWorkerApplicationBuilder ConfigureCosmosDB(this IFunctionsWorkerApplicationBuilder builder)
+        {
+            if (builder is null)
+            {
+                throw new System.ArgumentNullException(nameof(builder));
+            }
+
+            // TODO: How to handle scenarios where this method is used by a C# function?
+            // Avoid double calling?
+            // Could we write an analyzer to detect this?
+
+            // Tested and this is working. Almost all F# projects have a hidden reference to the FSharp.Core assembly
+            if (!IsFSharp(Assembly.GetCallingAssembly()))
```

Comment:
Instead of checking for F#, there are several ways to make this safe to call redundantly (and it is great practice to do so for bootstrap code!)

First evaluate the impact of running multiple times. When I look at `CosmosExtensionStartup` we can break down what happens:

1. Calls `AddAzureClientsCore()`. This is safe to call multiple times, it uses `TryAdd*` methods. So repeated calls are no-op.
2. Options configuration. Redundant calls _will_ cause these to be re-registered, but what is the impact of that? The implementations are safe to call multiple times. So, this is just sub-optimal. We can probably live with that.

If you _really_ want to solve the second part, you can use `TryAddEnumerable` overload and move the `IOptions<WorkerOptions>` binding logic into `CosmosDBBindingOptionsSetup`.

Reply:
Thanks for the insight! Does this look right? https://github.com/Azure/azure-functions-dotnet-worker/pull/1843/commits/86a04513d541c40ce27027482fe05aa7523b4c6e